storage/innobase/lock/lock0lock.cc

```diff
@@ -6428,3 +6443,86 @@ void lock_update_split_and_merge(
                           PAGE_HEAP_NO_SUPREMUM,
                           lock_get_min_heap_no(right_block));
 }
+#ifdef WITH_WSREP
+/*
+  Wsrep BF watchdog implementation:
+
+  We use InnoDB monitor thread to call wsrep_run_BF_lock_wait_watchdog()
+  periodically to check if there are applier threads hanging in
+  lock wait. If the lock wait has taken more than 5 seconds, the
+  potential local blocking transactions are killed until the
+  applier thread is released.
+ */
+
+std::atomic<uint> wsrep_BF_waiting_count;
```

Comment:
The variable is not documented by any a comment. Why is it not declared `static`?